Automattic/wp-calypso
# Site Settings: Remove sites-list usage from Analytics form

This PR removes the usage of `sites-list` from the Analytics settings form in favor of using the currently selected site. This also allows us to totally get rid of the `SiteSettingsAnalytics` wrapper component, as it becomes totally unnecessary. In an ideal world we'll be able to remove the `sites-list` dependency completely from the site settings controller, and this is a small step towards that.

To test:

* Checkout this branch
* Go to `/settings/analytics/$site` where `$site` is a Jetpack site with a Professional plan.
* Verify the form works as expected and there are no regressions there.
* Go to `/settings/analytics/$site` where `$site` is a Jetpack site with a Free plan.
* Verify the form is disabled and there are no regressions there.
* Go to `/settings/analytics/$site` where `$site` is a WordPress.com site with a Business plan.
* Verify the form works as expected and there are no regressions there.
* Go to `/settings/analytics/$site` where `$site` is a WordPress.com site with a Free plan.
* Verify the form is disabled and there are no regressions there.
